api/routes/index.js

```diff
@@ -44,10 +44,37 @@ router.get('/coop/:CoopID', async (req, res) => {
   }
 });
 
-//retrieving ALL co-ops in our coops table
+//retrieving ALL tags in our tags table and returning a list
+router.get('/tags', async (req, res) => {
+  try {
+    const query = await db.query(`SELECT * FROM tags;`);
+    res.send(query.rows);
+  } catch (error) {
+    console.log(error.stack);
+  }
+});
+
+//retrieve all coops with their tag
 router.get('/coops', async (req, res) => {
   try {
-    const query = await db.query(`SELECT * FROM coops;`);
+    const query = await db.query(`SELECT ARRAY(SELECT tag_name FROM coop_tags JOIN tags ON coop_tags.tag_id = tags.id WHERE coop_tags.coop_id= coops.id) AS tags,
+    * FROM coops;`);
+    res.send(query.rows);
+  } catch (error) {
+    console.log(error.stack);
+  }
+});
+
+//retrieve all coops with their tag
+router.get('/filteredCoops', async (req, res) => {
+  const tagParams = req.query.tags;
+
+  try {
+    const query = await db.query(
+      `SELECT ARRAY(SELECT tag_name FROM coop_tags JOIN tags ON coop_tags.tag_id = tags.id WHERE coop_tags.coop_id= coops.id) AS tags,
+    * FROM coops WHERE ARRAY(SELECT tag_id FROM coop_tags JOIN tags ON coop_tags.tag_id = tags.id WHERE coop_tags.coop_id= coops.id) @> $1;`,
```

Comment:
excellent sql magic 🧙 🧙‍♂️ 🧙‍♀️ 

nit: feel free to ignore until you finish the other task, but you can probably make this cleaner since you use the `SELECT tag_id FROM coop_tags JOIN tags ON coop_tags.tag_id = tags.id WHERE coop_tags.coop_id= coops.id` subquery twice

Reply:
they're,,, very similar,,, but not exactly the same,,,,,, so idk if we can do it

Reply:
got it no worries then, but can you style the query string so it's a bit more understandable though? see https://www.sqlstyle.guide/